frontend/src/utils/payment-links.ts

```diff
@@ -56,6 +58,7 @@ export async function generatePaymentLink({
   if (to) url.searchParams.set('to', to);
   if (token) url.searchParams.set('token', token.symbol);
   if (amount) url.searchParams.set('amount', amount);
+  if (chainId) url.searchParams.set('chainId', chainId.toString());
```

Comment:
Similar to the other BigNumber comment
```suggestion
  if (chainId) url.searchParams.set('chainId', BigNumber.from(chainId).toString());
```